File: processus/exo2.py
import os
import sys

pid1 = os.fork()
if pid1 < 0:
    print("Erreur lors de la creation du fils 1")
    sys.exit(1)
if pid1 == 0:
    print("Je suis le fils 1.")

pid2 = os.fork()
if pid2 < 0:
    print("Erreur lors de la creation du fils 2")
    sys.exit(1)
if pid1 == 0:
    print("Je suis le fils 2.")

os.waitpid(pid1, 0)
os.waitpid(pid2, 0)

print("Je suis le pere")

File: exo2.c
#include<stdio.h>
#include<unistd.h>
#include<stdlib.h>
#include<sys/types.h>
#include<sys/wait.h>

int main()
{
    pid_t pid1 = fork();
    
    if( pid1 == -1 )
    {
        puts("Erreur lors de la creation du fils 1.");
        exit(1) ;
    }
    if( pid1 == 0 )
    {   
        printf("Je suis le fils 1\n");
        exit(0);
    }
    
    pid_t pid2 = fork();
    if( pid2 == -1 )
    {
        puts("Erreur lors de la creation du fils 2.");
        exit(1) ;
    }
    if( pid2 == 0 )
    {   
        printf("Je suis le fils 2\n");
        exit(0);
    }
    wait(NULL);
    wait(NULL);

    printf("Je suis le pere.\n");
    return 0;
}